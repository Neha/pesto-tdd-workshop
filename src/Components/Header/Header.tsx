import React from 'react';

interface props {
    title: string
}
const headingStyle = {
    'backgroundColor' : 'rgb(220, 220, 220)',
    'color': 'rgb(0,0,0)'
}
const Header : React.FC<props> = ({ title }) => {
    return(
        <header data-testid="heading" style={headingStyle}>
            <h1 className="m-18">{title}</h1>
        </header>
    )
}

export default Header;
