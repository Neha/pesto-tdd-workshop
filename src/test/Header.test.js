import { render, screen } from '@testing-library/react';
import Header from '../Components/Header/Header';

test('renders Header', () => {
  const message = "Get a Quote";
  render(<Header title={message}/>);
  const headingText = screen.getByRole("heading", { name : message, level: 1})
  expect(headingText).toBeInTheDocument();
});

// test('should render red background color for Heading', () => {
//   const message = "Your heading comes here..."
//   render(<Header title={message}/>);
//   const heading = screen.getByTestId('heading');
//   expect(heading).toHaveStyle('background-color: rgb(220, 220, 220); color: rgb(0, 0, 0)');
// })






