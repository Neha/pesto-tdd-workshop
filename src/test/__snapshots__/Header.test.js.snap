// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> component Should match snapshot 1`] = `
<header
  data-testid="heading"
  style={
    Object {
      "backgroundColor": "rgb(220, 220, 220)",
      "color": "rgb(0,0,0)",
    }
  }
>
  <h1
    className="m-18"
  >
    Get a Quote
  </h1>
</header>
`;
